blueocean-dashboard/src/main/js/creation/bitbucket/cloud/BbCloudFlowManager.js

```diff
@@ -0,0 +1,414 @@
+            });
+        } else {
+            this.renderStep({
+                stateId: STATE.ERROR_UNKOWN,
+                stepElement: <BbUnknownErrorStep message={response.error} />,
+            });
+        }
+    }
+
+    @action
+    selectOrganization(organization) {
+        this.selectedOrganization = organization;
+        this.renderStep({
+            stateId: STATE.PENDING_LOADING_ORGANIZATIONS,
+            stepElement: <BbLoadingStep />,
+            afterStateId: STATE.STEP_CHOOSE_ORGANIZATION,
+        });
+        this._loadAllRepositories(this.selectedOrganization);
+    }
+
+    @action
+    selectRepository(repo) {
+        this.selectedRepository = repo;
+        this.pipelineName = this.selectedRepository.name;
+    }
+
+    @action
+    _loadAllRepositories(organization) {
+        this.repositories.replace([]);
+        this.repositoriesLoading = true;
+
+        let promise = null;
+        const cachedRepos = this._repositoryCache[organization.key];
+
+        if (cachedRepos) {
+            promise = new Promise(resolve => resolve({ repositories: { items: cachedRepos } }));
+        } else {
+            promise = this._loadPagedRepository(organization.key, FIRST_PAGE);
+        }
+
+        promise
+            .then(waitAtLeast(MIN_DELAY))
+            .then(repos => this._updateRepositories(organization.key, repos))
+            .catch(error => console.log(error));
+    }
+
+    _loadPagedRepository(organizationName, pageNumber, pageSize = PAGE_SIZE) {
+        return this._creationApi.listRepositories(this.credentialId, this.getApiUrl(), organizationName, pageNumber, pageSize);
+    }
+
+    @action
+    _updateRepositories(organizationName, repoData) {
+        const { items, nextPage } = repoData.repositories;
+        const firstPage = this.repositories.length === 0;
+        const morePages = !isNaN(parseInt(nextPage, 10));
+
+        this.repositories.push(...items);
+        this._repositoryCache[organizationName] = this.repositories.slice();
+
+        if (morePages) {
+            this._loadPagedRepository(organizationName, nextPage)
+                .then(repos2 => this._updateRepositories(organizationName, repos2, nextPage));
+        } else {
+            this.repositoriesLoading = false;
+        }
+        if (firstPage) {
+            // render the repo list only once, after the first page comes back
+            // otherwise we'll lose step's internal state
+            this.renderStep({
+                stateId: STATE.STEP_CHOOSE_REPOSITORY,
+                stepElement: <BbRepositoryStep />,
+                afterStateId: STATE.STEP_CHOOSE_ORGANIZATION,
+            });
+        }
+    }
+
+    saveRepo() {
+        this._saveRepo();
+    }
+
+    @action
+    _saveRepo() {
+        const afterStateId = this.isStateAdded(STATE.STEP_RENAME) ?
+            STATE.STEP_RENAME : STATE.STEP_CHOOSE_REPOSITORY;
+
+        this.renderStep({
+            stateId: STATE.STEP_COMPLETE_SUCCESS,
```

Comment:
This is a nit. Rather than immediately putting this into the `STATE.STEP_COMPLETE_SUCCESS` state, we should probably use `STATE.PENDING_CREATION_SAVING`. In `BbCloudCompleteStep.render` we should check for this state and if so set `loading=true` and then pass that down to the `FlowStep`. You can see the same pattern in `GithubCompleteStep` - it just makes the UI a little smoother and avoids a scenario where we briefly show success state but switch to error state after getting an error from backend.